sphinx/search/__init__.py

```diff
@@ -180,6 +180,16 @@ def __init__(self, document, lang):
         self.found_title_words = []
         self.lang = lang
 
+    def is_meta_keywords(self, node, nodetype):
+        is_meta = str(nodetype) == '<class \'sphinx.addnodes.meta\'>'
+        if is_meta and node.get('name', None) == 'keywords':
```

Comment:
This block is a bit complicated.

I think following code is a little simple and I prefer it::
```
if isinstance(node, sphinx.addnodes.meta) and node.get('name') == 'keywords':
    meta_lang = node.get('lang')
    if meta_lang is None:  # lang not specified
        return True
    elif meta_lang == self.lang.lang:  # matched to html_search_language
        return True

return False
```